x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/AbstractTrigonometricFunction.java

```diff
@@ -27,6 +29,10 @@ abstract class AbstractTrigonometricFunction extends UnaryScalarFunction {
         super(source, field);
     }
 
+    protected AbstractTrigonometricFunction(StreamInput in) throws IOException {
```

Comment:
Is this actually necessary? The concrete classes will still need a `StreamInput` constructor that just calls the super constructor, and it can skip over this layer entirely.

Reply:
I don't believe you can "skip" layers like that.